packages/react-components/src/components/chart/hooks/useResize.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useRef } from 'react';
+import { calculateTrendCursorsSeriesMakers, calculateXFromTimestamp } from '../utils/getInfo';
+import { onResizeUpdateTrendCursorYValues } from '../utils/getTrendCursor';
+import { TrendCursorProps } from '../types';
+
+const useResize = ({ size, series, yMin, yMax, graphic, setGraphic, viewport }: TrendCursorProps) => {
+  const prevSize = useRef(size);
+  useEffect(() => {
```

Comment:
Why can't you use a resize event instead of a side-effect?

Reply:
are you referring to the resize of the echarts ? if so, that only handles the out of the box features i.e resizing of the series lines x and y axis etc... given TCs are custom, we need to also handle the resize event